games-board/chessx/chessx-1.4.6.recipe

```diff
@@ -0,0 +1,88 @@
+SUMMARY="A free and open source chess database application"
+DESCRIPTION="With ChessX you can operate your collection of chess games \
+in many ways: browse, edit, add, organize, analyze, etc."
+HOMEPAGE="https://chessx.sourceforge.net/"
+COPYRIGHT="2005-2016 ChessX developers
+	1990-2000 Info-ZIP
+	1998-2005 Gilles Vollant
+	1999-2004 Shane Hudson
+	2003 Sune Fischer
+	2005 Kamil Przybyla
+	2005-2006 Ejner Borgbjerg
+	2005-2006 William Hoggarth
+	2005-2007 Marius Roets
+	2005-2010 Michal Rudolf
+	2005-2011 Sergey A. Tachenov
+	2006 Tobias Koenig
+	2006-2007 Pascal Georges
+	2006-2007 Sean Estabrooks
+	2007 Heinz R. Hopfgartner
+	2007 Rico Zenklusen
+	2010 P. Sereno
+	2011 Nokia Corporation and/or its subsidiary(-ies).
+	2012-2016 Jens Nissen
+	2014 Digia Plc and/or its subsidiary(-ies).
+	2014 Markus Kolb"
+LICENSE="GNU GPL v2
+	BSD (3-clause)
+	GNU LGPL v2
+	GNU LGPL v2.1
+	Digia Qt LGPL Exception v1.1"
+REVISION='1'
+SOURCE_URI="https://downloads.sourceforge.net/chessx/chessx-$portVersion.tgz"
+CHECKSUM_SHA256="ad951910a3bd53f5655f9e85101e7969b536078473e09a2a7dded6ef171a68ed"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	chessx = $portVersion
```

Comment:
```diff
 PROVIDES="
-	chessx = $portVersion
+	chessx$secondaryArchSuffix = $portVersion
-	app:chessx = $portVersion
+	app:chessx$secondaryArchSuffix = $portVersion
 	"
```
Does anyone know if it's correct to also add the `$secondaryArchSuffix` after `app:chessx`?

Reply:
@fbrosson Doesn't [look][0] like so to me

[0]: https://github.com/haikuports/haikuporter/blob/2b7ba23e93e799bb9ceeab552266cc1e35b26c5f/HaikuPorter/Port.py#L1205

Reply:
I might be wrong but this does not seem to be related to the `app:*` elements but to the `$appsDir` path only.

Well, I just did some metrics (using find, grep, sort, cut and comm, which are great for that as we know :)

Total number of recipes with an `app:` and 2nd arch | 132
--- | ---
`app:` **_with_** `$secondaryArchSuffix` | 78
`app:` **_without_** `$secondaryArchSuffix` | 54

BTW, it's funny, because [qemacs](https://github.com/haikuports/haikuports/tree/master/app-editors/qemacs) has the `$secondaryArchSuffix` for its `app:`, while [xemacs](https://github.com/haikuports/haikuports/tree/master/app-editors/xemacs) does not.

There does not seem to be a real trend, but I think it would be safer to recommend using the `$secondaryArchSuffix` for `app:` elements.

(In my metrics above I did not count my "copy" of chessx.)

Reply:
The $secondaryArchSuffix makes sense only if the app is available both for the secondary and primary archs at the same time. This is an unlikely case. If the app is only available for one, then it makes sense to not use the $secondaryArchSuffix. This way, people can "pkgman install app:chessx" and not worry about which compiler was used.

Reply:
OK, thanks for the explanation. Now I understand why there is no real trend.

I do agree that, on x86_gcc2, we would not be able to run `pkgman install app:chessx`, but with a simple `pkgman search app:chessx` we would get the answer.
For instance, this is what we get with notepadqq on x86_gcc2:
```
pkgman search app:notepadqq
Status  Name           Description                        
----------------------------------------------------------
        notepadqq_x86  Multiplatform Notepad++-like editor
```

IMO there is something worse than having to call to `pkgman install app:notepadqq_x86` instead of just `pkgman install app:notepadqq`:
It's having to remember that if your first attempt did not succeed, then this could be because the package you are trying to install... has the suffix on its `app:`.

Don't you agree that adding the suffix increases the consistency?

BTW, among the 78 recipes which already have the suffix for their `app:`, 65 are broken on x86_gcc2 primary arch, 4 are marked as untested (on that same arch), and 7 are marked as tested (also on that same arch.)
So, among those 78 recipes, the only which would be correctly using the suffix, according to your rationale, would be:
* these 4 recipes, which are marked as untested:
```
kde-base/kolourpaint/kolourpaint-17.11.80.recipe
kde-base/kstars/kstars-2.8.8.recipe
media-gfx/krita/krita-3.3.1.recipe
media-gfx/nomacs/nomacs-3.8.0.recipe
```
* and these 7 other recpes, which are marked as tested:
```
app-editors/qemacs/qemacs-0.4.1~pre20170225.recipe
games-puzzle/fish-fillets/fish_fillets-1.0.1.recipe
games-strategy/homeworld_sdl/homeworld_sdl-0.6.recipe
haiku-apps/cdplayer/cdplayer-1.0.recipe
haiku-apps/clockwerk/clockwerk-1.0~git.recipe
haiku-apps/paladin/paladin-1.4~git.recipe
haiku-games/puri/puri-0.3.9.1.recipe
haiku-games/superfreecell/superfreecell-0.1.0.recipe
mail-client/mailnews/mailnews-2.0.0.25.recipe
```
**All** the other **65** recipes (which have the suffix for their `app:` elements) are **broken** on x86_gcc2 primary arch (and yet have the suffix for their `app:` elements.)

Now, in my previous table I wrote that among those recipes which have support for 2nd arch, there are only 54 recipes without the suffix for the `app:` elements, which is less than the number of recipes that do not follow your rationale.

Since your argument is beaten by the metrics above, wouldn't it be easier to just decide that, from now on, all new recipes (with support for secondary arch) should have the suffix on their `app:` elements?

Please don't get me wrong. I'm not trying to fight anyone. You are here since much longer than me, and you also know Haiku and HaikuPorts much better than me.
I'm just trying to shed some light on a (small) inconsistency to improve things a little. Thanks!

Reply:
As I said, having an app exist on both arches is a *very* rare case. The only possible reason would be that the app uses add-ons, and you want to be able to use different add-ons that are not all available for gcc2 or for gcc5. I am not aware of any app where this is the case.

So I think the better thing to do is make sure all "app:" recipe are enabled for only one of either x8-_gcc2 (primary) or x86 (secondary), and that they never have a $secondaryArchSuffix. Or if one of them has a reason for having one, it should be written in the recipe in big red letters (ok, we can't do that) in a comment next to it.

Reply:
OK.
I'm not convinced that droping the suffix for `app:chessx` is a good idea, but can live with that :)
(I'll update my previous review.)

P.S. I guess the reason why we have so many recipes with the suffix, although they are broken on x86_gcc2) is that there are more contributors who think the suffix should be used than contributors who think it shouldn't. But I do admit that this is not a valid argument :)

Reply:
Maybe we should explain when to use and when not to use it. I think people think the suffix is mandatory everywhere when using the secondary arch (if that was the case, we wouldn't require recipe writers to add it and the tools would do it on their own).

The $secondaryArchSuffix is used to make sure the entries for primary and secondary arch do not conflict wit each other. This makes sense only if all these conditions are true:
- The recipe builds for both primary and secondary arch
- The result of the build is different in each case and it makes sens to have both versions installed

In all other cases, the suffix should not be used.
For example, the ffmpeg recipe builds for both architectures (in different versions) and provide libraries with the secondaryArchSuffix. But, only the x86 package provides binaries (ffplay, ffmpeg, ffprobe, ...). So, these binaries are provided without $secondaryArchSuffix.

Reply:
Guess what? You have now convinced me, and I have even found one more argument for you ;)

When we use the commandBinDir "trick", we set commandSuffix to the empty string, and we do so because the `cmd:` elements are in `$prefix/bin` and don't have the suffix (even when wr're on x86_gcc2, since this is the purpose of the commandBinDir trick.)

So we just need to say that `app:` elements follow the same rules as the `cmd:` elements.
Since we are providing `$appsDir/chessx` (and not `$appsDir/chessx_x86`), it is perfectly normal to declare `app:chessx` **without** any suffix.

So congratulations, and thanks for having convinced me!
That wasn't easy, sorry ;D